lib/wasi/src/os/task/process.rs

```diff
@@ -232,7 +231,7 @@ impl WasiProcess {
             if self.waiting.load(Ordering::Acquire) > 0 {
                 let mut triggered = false;
                 for pid in children.iter() {
-                    if let Some(process) = self.compute.get_process(*pid) {
+                    if let Some(process) = self.compute.must_upgrade().get_process(*pid) {
```

Comment:
is that ok to use the `must_upgrade` in an `if`? What happens if the if is not true?

Reply:
It will panic. 

Which should be perfectly fine, because this can only happen if you completely misuse the API.

`WasiControlPlane` is stored on `WasiEnv`, so this can never fail as long as the `WasiEnv` still exists. 
(Unless someone was to *replace* the WasiControlPlane , but that would also be serious misuse and worthy of a panic. )